src/common/util/collision.js

```diff
@@ -0,0 +1,31 @@
+const collisionHelper = (node, nodeToCheck, filter, collisions = []) => {
+    if (!filter || (filter(node) && node.id !== nodeToCheck.id)) {
+        const node1LeftX = node.pos.x;
+        const node1RightX = node.pos.x + node.size.x;
+        const node2LeftX = nodeToCheck.pos.x;
+        const node2RightX = nodeToCheck.pos.x + nodeToCheck.size.x;
+
+        const node1TopY = node.pos.y;
+        const node1BottomY = node.pos.y + node.size.y;
+        const node2TopY = nodeToCheck.pos.y;
+        const node2BottomY = nodeToCheck.pos.y + nodeToCheck.size.y;
+
+        const oneToTheLeft = node2RightX < node1LeftX || node1RightX < node2LeftX;
+        const oneBelow = node1TopY > node2BottomY || node2TopY > node1BottomY;
+        if (!(oneToTheLeft || oneBelow)) {
```

Comment:
I think we need a more complex algorithm to find collisions, consider: https://gyazo.com/bde59d1bf5fbf0f977520388eb8f5600

in that case we would report back no collision using this function (I think)

Reply:
I tested locally with this:
```
const { checkCollisions } = require('./src/common/util');
const { GameNode } = require('squishjs');

const thingOne = GameNode([255, 0, 0, 255], null, {x: 0, y: 0}, {x: 100, y: 100});
const thingTwo = GameNode([0, 0, 255, 255], null, {x: 50, y: 50}, {x: 25, y: 25});

const wouldBeCollisions = checkCollisions(thingOne, thingTwo);
console.log(wouldBeCollisions);
```
The output was 
```
[
  GameNode {
    id: 0,
    children: [],
    color: [ 255, 0, 0, 255 ],
    handleClick: null,
    pos: { x: 0, y: 0 },
    size: { x: 100, y: 100 },
    text: undefined,
    asset: undefined,
    listeners: Set {},
    playerId: 0
  }
]
```
I found this algorithm online somewhere and the math _seems_ correct to me as a plebeian. Maybe we need to write a proof 🤔